pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -262,6 +267,19 @@ private void extractH3IndexConfigsFromTableConfig(TableConfig tableConfig) {
     }
   }
 
+  private void extractNoDictionaryColumnCompressionCodecConfigsFromTableConfig(TableConfig tableConfig) {
+    List<String> rawIndexColumns = new ArrayList<>();
+    List<FieldConfig> fieldConfigList = tableConfig.getFieldConfigList();
+    if (fieldConfigList != null) {
+      for (FieldConfig fieldConfig : fieldConfigList) {
+        if (fieldConfig.getEncodingType() == FieldConfig.EncodingType.RAW && fieldConfig.getNoDictionaryColumnCompressorCodec() != null) {
+          rawIndexColumns.add(fieldConfig.getName());
+        }
+      }
+    }
+    setRawIndexColumnCompressionType(tableConfig.getIndexingConfig(), rawIndexColumns);
```

Comment:
This doesn't seem right. At line 197 it will check if the map is null or not. If null, then we won't be able to set ZSTD/SNAPPY coming via FieldConfig (from this function). Right ?

Reply:
so, if the noDictinaryColumnMap is empty/null this indicates that no column name to compressionType is set right? So you are right, the field config map will not be set here.

Reply:
I think we should not use the old method here. Let it be there. Just remove the call to it at line 280

Instead, we can probably do the following. Wdyt ?

`_rawIndexCreationColumns.add(fieldConfig.getName())` at line 276

`_rawIndexCompressionType.put(fieldConfig.getName(), ChunkCompressionType.valueOf(fieldConfig.getNoDictionaryColumnCompressorCodec())`

Reply:
In any case, please make sure to run the end-to-end query execution test in debug mode and ensure that segment generation code is correctly picking up the compression codec config

Reply:
updated!